gobblin-runtime/src/main/java/org/apache/gobblin/runtime/spec_catalog/FlowCatalog.java

```diff
@@ -360,16 +361,21 @@ public Map<String, AddSpecResponse> put(Spec spec, boolean triggerListener) {
         responseMap.put(entry.getKey().getName(), entry.getValue().getResult());
       }
     }
+    AddSpecResponse<String> schedulerResponse = responseMap.getOrDefault(ServiceConfigKeys.GOBBLIN_SERVICE_JOB_SCHEDULER_LISTENER_CLASS, new AddSpecResponse<>(null));
 
-    if (isCompileSuccessful(responseMap)) {
+    if (isCompileSuccessful(schedulerResponse.getValue())) {
```

Comment:
When it exceed quota, why compile is marked as succeed?It's a little confuse to read the code

Reply:
Maybe there's a better terminology here:
For compilation how I interpret it is that the flow configuration can compile (src and destination and any required parameters exist).
But it can pass the compilation step but fail on a resource validation check, which doesn't mean that the flow was improperly compiled or that the inputs were incorrect. It's more that the users have too many flows already sent in the system.

Reply:
Can we add comments to explain this? Just to make the code more easy to read

Reply:
a large part of this is the tunneling I describe above, but if you actually do need additional info to support `isExplain/hasExplain`, you could insert it into (or wrapping around) the `QuotaExceededException`